packages/lib/src/project/AppProject.ts

```diff
@@ -165,53 +166,65 @@ export default class AppProject extends BasePackageProject {
     return this.version;
   }
 
-  // TODO: Testme
-  public async getImplementation({ packageName, contractName }: { contractName: string, packageName?: string }): Promise<string> {
-    return this.app.getImplementation(packageName || this.name, contractName);
+  public async getImplementation({ packageName, contractName, contract }: { contractName?: string, packageName?: string, contract?: Contract }): Promise<string> {
+    return this.app.getImplementation(packageName || this.name, contractName || contract.schema.contractName);
   }
 
-  // TODO: Testme
   public async createContract(contract: Contract, { packageName, contractName, initMethod, initArgs }: ContractInterface = {}): Promise<Contract> {
     if (!contractName) contractName = contract.schema.contractName;
     if (!packageName) packageName = this.name;
     return this.app.createContract(contract, packageName, contractName, initMethod, initArgs);
   }
 
   public async createProxy(contract: Contract, contractInterface: ContractInterface = {}): Promise<Contract> {
-    const { contractName, packageName, initMethod, initArgs } = this.getContractInterface(contract, contractInterface);
-    const proxyAdmin = await this.ensureProxyAdmin();
-    return this.app.createProxy(contract, packageName, contractName, proxyAdmin.address, initMethod, initArgs);
+    const { contractName, packageName, initMethod, initArgs, admin } = this.getContractInterface(contract, contractInterface);
+    const proxyAdmin = admin || (await this.ensureProxyAdmin()).address;
+    return this.app.createProxy(contract, packageName, contractName, proxyAdmin, initMethod, initArgs);
   }
 
-  protected getContractInterface(contract: Contract, { contractName, packageName, initMethod, initArgs }: ContractInterface = {}): ContractInterface {
+  protected getContractInterface(contract: Contract, opts: ContractInterface = {}): ContractInterface {
+    let { contractName, packageName, initMethod } = opts;
+
     if (!contractName) {
       contractName = contract.schema.contractName;
     }
     if (!packageName) {
       packageName = this.name;
     }
-    if (!isEmpty(initArgs) && !initMethod) {
+    if (!isEmpty(opts.initArgs) && !initMethod) {
       initMethod = 'initialize';
     }
-    return { contractName, packageName, initArgs, initMethod };
+
+    return { ...opts, contractName, packageName, initMethod };
   }
 
-  public async createProxyWithSalt(contract: Contract, salt: string, contractInterface: ContractInterface = {}): Promise<Contract> {
-    const { contractName, packageName, initMethod, initArgs } = this.getContractInterface(contract, contractInterface);
+  public async createProxyWithSalt(contract: Contract, salt: string, signature?: string, contractInterface: ContractInterface = {}): Promise<Contract> {
```

Comment:
I believe this is not how conditional arguments should be used (as a matter of fact, if you don't set a defualt value to `contractIinterface`, tsc will fail) They should be defined always at the end, so you can avoid passing `null` in calls. 
As this may lead to future errors, I would reorder the parameters as: `contract`, `contractInterface`, `salt` and `signature?`. WDYT?

Reply:
I'd agree, but contractInterface is also optional here (since it has a default value), so it's ok to flag signature as optional.